python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -339,13 +351,8 @@ def run(self, science, matchedTemplate, difference,
         if idFactory is None:
             idFactory = lsst.meas.base.IdGenerator().make_table_id_factory()
 
-        # Ensure that we start with an empty detection and deblended mask.
-        mask = difference.mask
-        clearMaskPlanes = ["DETECTED", "DETECTED_NEGATIVE", "NOT_DEBLENDED", "STREAK"]
-        for mp in clearMaskPlanes:
-            if mp not in mask.getMaskPlaneDict():
-                mask.addMaskPlane(mp)
-        mask &= ~mask.getPlaneBitMask(clearMaskPlanes)
+        sigma = self._prepareInputs(difference)
+        self.log.info("Using PSF sigma of %f to smooth the difference image for detection.", sigma)
```

Comment:
This new log is worth keeping, but I think it should go into `_prepareInputs`, since you forgot to put this log after the other place and there's no need to pass `sigma` to detection since it will get it from the (now guaranteed valid) psf.

Reply:
I purposely left it out of `_prepareInputs`, because that is used by both regular diffim and preconvolution where it is incorrect (since we don't smooth the difference image for detection then)